pandas/_libs/groupby.pyx

```diff
@@ -1411,7 +1407,7 @@ def group_min(groupby_t[:, :] out,
 
         for i in range(ncounts):
             for j in range(K):
-                if nobs[i, j] == 0:
+                if nobs[i, j] < min_count or nobs[i, j] == 0:
```

Comment:
can you avoid the double lookup/comparison here by setting `min_count = max(min_count, 1)` outside the loop?

Reply:
Done